packages/clerk-js/src/ui/components/SessionTasks/tasks/TaskResetPassword/index.tsx

```diff
@@ -0,0 +1,188 @@
+import { useClerk, useReverification } from '@clerk/shared/react';
+import type { UserResource } from '@clerk/shared/types';
+import { useCallback } from 'react';
+
+import { useEnvironment, useSignOutContext, withCoreSessionSwitchGuard } from '@/ui/contexts';
+import { Col, descriptors, Flow, localizationKeys, useLocalizations } from '@/ui/customizables';
+import { Card } from '@/ui/elements/Card';
+import { useCardState, withCardStateProvider } from '@/ui/elements/contexts';
+import { Form } from '@/ui/elements/Form';
+import { Header } from '@/ui/elements/Header';
+import { useConfirmPassword } from '@/ui/hooks';
+import { useMultipleSessions } from '@/ui/hooks/useMultipleSessions';
+import { handleError } from '@/ui/utils/errorHandler';
+import { createPasswordError } from '@/ui/utils/passwordUtils';
+import { useFormControl } from '@/ui/utils/useFormControl';
+
+import { withTaskGuard } from './withTaskGuard';
+
+const TaskResetPasswordInternal = () => {
+  const { signOut, user, session } = useClerk();
+  const card = useCardState();
+  const {
+    userSettings: { passwordSettings },
+  } = useEnvironment();
+
+  const { t, locale } = useLocalizations();
+  const { otherSessions } = useMultipleSessions({ user });
+  const { navigateAfterSignOut, navigateAfterMultiSessionSingleSignOutUrl } = useSignOutContext();
+  const updatePasswordWithReverification = useReverification(
+    (user: UserResource, opts: Parameters<UserResource['updatePassword']>) => user.updatePassword(...opts),
+  );
+
+  const handleSignOut = () => {
+    if (otherSessions.length === 0) {
+      return signOut(navigateAfterSignOut);
+    }
+
+    return signOut(navigateAfterMultiSessionSingleSignOutUrl, { sessionId: session?.id });
+  };
```

Comment:
Not a blocker for this PR, we can do it in a separate one. I have a proposal to move this sign-out with multi-session handling within the `SignOutContext` as we're duplicating between two task components now. 

We'd have to double-check on how it's being handled within the `UserButton` popover and potentially reuse it there.